esiclient/tests/functional/v1/test_node.py

```diff
@@ -137,3 +139,180 @@ def test_admin_can_node_power_on_off(self):
+
+        if 'test_node_ident' not in self.config['functional']:
+            self.fail('test_node_ident must be specified in test config')
+        self.node = self.config['functional']['test_node_ident']
+
+    def test_admin_can_enable_console(self):
+        """Tests admin console functionality.
+
+        Tests that an admin project which is not set to a node's 'owner'
+        nor 'lessee' can perform the steps to setup an ipmitool-socat
+        serial console.
+
+        Test steps:
+            1) Check that the project can run command
+                `baremetal node console enable <node>`
+            2) Check that project can run command
+                baremetal node console show <node>`
+            3) Check that project can run command
+                `baremetal node console disable <node>`
+            4) (cleanup) disable console
+        """
+
+        node = utils.node_show(self.clients['admin'], self.node)
+        if node['provision_state'] != 'manageable':
+            utils.node_set_provision_state(self.clients['admin'],
+                                           self.node, 'manage')
+
+        utils.node_console_enable(self.clients['admin'], self.node)
+        self.addCleanup(utils.node_console_disable,
+                        self.clients['admin'],
+                        self.node,
+                        fail_ok=True)
+        time.sleep(10)
+        console = utils.node_console_show(self.clients['admin'], self.node)
+        assert console['console_enabled']
+
+        utils.node_console_disable(self.clients['admin'], self.node)
+        time.sleep(10)
+        console = utils.node_console_show(self.clients['admin'], self.node)
+        assert not console['console_enabled']
+
+    def test_owner_can_enable_console(self):
+        """Tests owner console functionality.
+
+        Tests that an owner can perform the steps to setup an ipmitool-socat
+        serial console.
+
+        Test steps:
+            1) Set owner of node to random project
+            2) Check that the owner can run command
+                `baremetal node console enable <node>`
+            3) Check that owner can run command
+                baremetal node console show <node>`
+            4) Check that owner can run command
+                `baremetal node console disable <node>`
+            5) (cleanup) Unset the owner
+            6) (cleanup) disable console
+        """
+
+        node = utils.node_show(self.clients['admin'], self.node)
+        if node['provision_state'] != 'manageable':
+            utils.node_set_provision_state(self.clients['admin'],
+                                           self.node, 'manage')
+
+        utils.node_set(self.clients['admin'],
+                       self.node,
+                       'owner', self.projects['random']['id'])
+        self.addCleanup(utils.node_set,
+                        self.clients['admin'],
+                        self.node, 'owner')
```

Comment:
You don't want to unset the owner; you want to set it to what it was previously by passing in node['owner']

Reply:
fixed